modules/EPMC.py

```diff
@@ -0,0 +1,183 @@
+    logging.info(f'Generating evidence:')
+
+    filtered_target_terms = ["TEC", "TECS", "Tec", "tec", "'", "(", ")", "-", "-S", "S", "S-", "SS", "SSS",
+         "Ss", "Ss-", "s", "s-", "ss", "U3", "U6", "u6", "SNORA70", "U2", "U8"]
+
+    ##
+    ## Load/filter datasets
+    ##
+    # partitionKeys = ['pmid', 'targetFromSourceId', 'diseaseFromSourceMappedId']
+    # w = Window.partitionBy(*partitionKeys)
+    filtered_cooccurrence_df = (
+        # Reading file:
+        spark.read.parquet(cooccurrenceFile)
+
+        # Filtering for diases/target cooccurrences:
+        .filter(
+            (col('type') == "GP-DS") & 
+            (col('isMapped') == True) & 
+            (col('pmid').isNotNull()) & 
+            (length(col('text')) < 600) &
+            (col('label1').isin(filtered_target_terms) == False)
+        )
+        # Renaming columns:
+        .withColumnRenamed("keywordId1", "targetFromSourceId")
+        .withColumnRenamed("keywordId2", "diseaseFromSourceMappedId")
+        .withColumnRenamed("label1", "targetFromSource")
+        .withColumnRenamed("label2", "diseaseFromSource")
+
+        .withColumn('tmp',col('pmid').cast(StringType()))
+    )
+
+    # Report on the number of diseases, targets and associations:
+    logging.info(f'Number of publications: {filtered_cooccurrence_df.select(col("tmp")).distinct().count()}')
+    logging.info(f'Number of targets: {filtered_cooccurrence_df.select(col("targetFromSourceId")).distinct().count()}')
+    logging.info(f'Number of diseases: {filtered_cooccurrence_df.select(col("diseaseFromSourceMappedId")).distinct().count()}')
+    logging.info(f'Number of associations: {filtered_cooccurrence_df.select(col("diseaseFromSourceMappedId"), col("targetFromSourceId")).dropDuplicates().count()}')
+
+    # Aggregating cooccurrence, get score apply filter:    
+    aggregated_df = (
+        filtered_cooccurrence_df
+
+        # Aggregating data by publication, target and disease:
+        .groupBy(['pmid', 'targetFromSourceId', 'diseaseFromSourceMappedId'])
+        .agg(
+            first(col("targetFromSource")).alias("targetFromSource"),
+            first(col("diseaseFromSource")).alias("diseaseFromSource"),
+            collect_set(col('tmp')).alias('literature'),
+            collect_set(
+                struct(  
+                    col("text"),
+                    col('start1').alias('tStart'),
+                    col("end1").alias('tEnd'),
+                    col('start2').alias('dStart'),
+                    col("end2").alias('dEnd'),
+                    col('section')
+                )
+            ).alias('textMiningSentences'),
+            sum(col('evidence_score')).alias('resourceScore')
+        )
+
+        # Summarizing all scores then filter for evidence string with at least score == 2
+        # .withColumn('resourceScore', sum(col('evidence_score')).over(w))
+        # .withColumn('resourceScores', collect_list(col('evidence_score')).over(w))
```

Comment:
Ah, I suppose this relates to the commented-out code above which defined the `w`. I suggest:
* If this is never required, simply remove it;
* If this is sometimes required, make it an argument parser option and invoke only if needed

Reply:
In an earlier version of the script we used windowing, but when I tried to use windows to summarize the score, it was failing (hence the comment in the beginning of the script about the memory issue)